core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -206,20 +206,75 @@ public void start() {
             AtomicInteger attempt = new AtomicInteger(0);
             Unreliables.retryUntilSuccess(startupAttempts, () -> {
                 logger().debug("Trying to start container: {} (attempt {}/{})", image.get(), attempt.incrementAndGet(), startupAttempts);
-                tryStart(profiler.startNested("Container startup attempt"));
+                tryRun(profiler.startNested("Container startup attempt"));
                 return true;
             });
-
         } catch (Exception e) {
             throw new ContainerLaunchException("Container startup failed", e);
         } finally {
             profiler.stop().log();
         }
     }
 
-    private void tryStart(Profiler profiler) {
+    /**
+     * Restarts a running container.
+     *
+     * @throws ContainerNotStartedException if {@link #start()} hasn't been called first
+     */
+    public void restart() {
+        Profiler profiler = new Profiler("Container restart");
+        profiler.setLogger(logger());
+
+        if (null == containerId) {
+            throw new ContainerNotStartedException("The container is not started: run start() first");
+        }
+
         try {
-            String dockerImageName = image.get();
+            profiler.start("Restarting running container");
+            logger().debug("Restarting container: {}", getDockerImageName());
+
+            AtomicInteger attempt = new AtomicInteger(0);
+            Unreliables.retryUntilSuccess(startupAttempts, () -> {
+                logger().debug("Trying to restart container: {} (attempt {}/{})", image.get(),
+                    attempt.incrementAndGet(), startupAttempts);
+                tryRestart(profiler.startNested("Container restart attempt"));
+                return true;
+            });
+        } finally {
+            profiler.stop().log();
+        }
+    }
+
+    private void tryRestart(Profiler profiler) {
```

Comment:
I don't think we need these private methods (`tryRestart`, `tryRun`), they make code looks like there is some logic behind it